content/blog/kubernetes-clusters-as-code-part1.md

```diff
@@ -0,0 +1,1094 @@
+
+The first lines of your Terraform configuration file should look like this:
+
+<!--EndFragment-->
+
+```
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+```
+
+<!--StartFragment-->
+
+</br>
+
+## Create a Cluster resource
+
+### Terraform data source for Cluster Blueprint
+
+In order to use the data source available for cluster blueprint, you should add the below block in your terraform file, and specify the cluster blueprint name. Using this data source, terraform will fetch the cluster Blueprint ID associated with it.
+
+In the below block, "demo" is the cluster blueprint **name** provided:
+
+<!--EndFragment-->
+
+```
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+```
+
+<!--StartFragment-->
+
+The cluster blueprint ID can now be fetched by using:
+
+<!--EndFragment-->
+
+```
+blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+```
+
+<!--StartFragment-->
+
+</br>
+
+### Terraform data source for Site
+
+In order to use the data source available for site, you should add the below block in your terraform file, and specify the site name. Using this data source, Terraform will fetch the site ID associated with it.
+
+In the below block, "BLR" is the site **name** provided:
+
+<!--EndFragment-->
+
+```
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = ""
+}
+```
+
+<!--StartFragment-->
+
+The site ID can now be fetched by using:
+
+<!--EndFragment-->
+
+```
+site_id = data.hpegl_caas_site.blr.id
+```
+
+<!--StartFragment-->
+
+</br>
+
+###  Terraform data source for Cluster
+
+In order to use the data source available for cluster, you should add the below block and provide the cluster name and space id. Using this data source, terraform will fetch the cluster server and user token associated with it.
+
+In the below block, "tf-test" is the **name** of the pre-created cluster.
+
+<!--EndFragment-->
+
+```
+data "hpegl_caas_cluster" "test" {
+  name     = "tf-test"
+  space_id = ""
+}
+```
+
+<!--StartFragment-->
+
+The **host** (cluster server) and **token** (user token) can now be fetched from the cluster kubeconfig using:
+
+<!--EndFragment-->
+
+```
+host     = yamldecode(base64decode(data.hpegl_caas_cluster.test.kubeconfig)).clusters[0].cluster.server
+token    = yamldecode(base64decode(data.hpegl_caas_cluster.test.kubeconfig)).users[0].user.token
+```
+
+<!--StartFragment-->
+
+</br>
+
+### Terraform resource for Cluster
+
+In order to create a cluster using the cluster resource, the following values should be specified in the **cluster-create.tf** file:
```

Comment:
I am a bit confused. Is the file cluster-create.tf a new file or is it the same file as the TF configuration file you created earlier?
If it is the latter, I do not see the "host" and "token" statements in the cluster-create.tf file.

Reply:
the host and token is a part of cluster **data-source** while cluster-create.tf is a cluster **resource.**